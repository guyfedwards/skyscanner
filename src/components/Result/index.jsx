import React from 'react';
import BpkCard from 'bpk-component-card';
import styled from 'styled-components';

import flightType from '../../types/flight';
import ResultFooter from '../ResultFooter';
import Leg from '../Leg';

const StyledCard = styled(BpkCard)`
  :not(last-of-type) {
    margin-bottom: 20px;
  }
`;

const Result = ({ flight }) => (
  <StyledCard>
    {flight.legs.map(l => (
      <Leg leg={l} key={l.id} />
    ))}
    <ResultFooter agent={flight.agent} price={flight.price} />
  </StyledCard>
);

Result.propTypes = {
  flight: flightType.isRequired,
};

export default Result;
