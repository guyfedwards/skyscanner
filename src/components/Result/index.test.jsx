import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Result from '.';

const flight = {
  id: 'it_1',
  legs: [
    {
      id: 'leg_1',
      departure_airport: 'BUD',
      arrival_airport: 'LTN',
      departure_time: '2020-10-31T15:35',
      arrival_time: '2020-10-31T17:00',
      stops: 0,
      airline_name: 'Wizz Air',
      airline_id: 'WZ',
      duration_mins: 145,
    },
  ],
  price: 'Â£35',
  agent: 'Wizzair.com',
  agent_rating: 9.1,
};

describe('Result', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Result flight={flight} />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Result flight={flight} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
