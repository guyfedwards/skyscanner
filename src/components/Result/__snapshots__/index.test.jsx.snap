// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render correctly 1`] = `
<ForwardRef(Styled(BpkCard))>
  <Leg
    leg={
      Object {
        "airline_id": "WZ",
        "airline_name": "Wizz Air",
        "arrival_airport": "LTN",
        "arrival_time": "2020-10-31T17:00",
        "departure_airport": "BUD",
        "departure_time": "2020-10-31T15:35",
        "duration_mins": 145,
        "id": "leg_1",
        "stops": 0,
      }
    }
  />
  <ResultFooter
    agent="Wizzair.com"
    price="Â£35"
  />
</ForwardRef(Styled(BpkCard))>
`;
